qtpy/QtCore.py

```diff
@@ -20,14 +20,27 @@
     from PyQt5.QtCore import QT_VERSION_STR as __version__
 
     # For issue #153 and updated for issue #305
-    from PyQt5.QtCore import QDate, QDateTime, QTime
     QDate.toPython = lambda self, *args, **kwargs: self.toPyDate(*args, **kwargs)
     QDateTime.toPython = lambda self, *args, **kwargs: self.toPyDateTime(*args, **kwargs)
     QTime.toPython = lambda self, *args, **kwargs: self.toPyTime(*args, **kwargs)
 
+    # Fix enums in PyQt5 5.9.*
+    from .enums_compat import demote_enums
+    if PYQT5 and __version__.startswith('5.9.'):
+        from PyQt5 import QtCore
+        demote_enums(QtCore)
+        del QtCore
+    del demote_enums
```

Comment:
```suggestion
    if __version__.startswith('5.9.'):
        from .enums_compat import demote_enums
        from PyQt5 import QtCore
        demote_enums(QtCore)
        del QtCore, demote_enums
```

* Move `.enums_compt` import and deletion inside the block, so it's only executed if nessesary
* Remove redundant `PYQT5` check (since we're already inside the `PYQT5` block)